ros2_demo.py

```diff
@@ -0,0 +1,362 @@
+#!/usr/bin/env python
+
+import os
+import time
+
+import numpy
+
+from pydrake.common import FindResourceOrThrow
+from pydrake.common.value import AbstractValue
+from pydrake.geometry import ConnectDrakeVisualizer
+from pydrake.geometry import FramePoseVector
+from pydrake.math import RigidTransform
+from pydrake.multibody.parsing import Parser
+from pydrake.systems.analysis import Simulator
+from pydrake.multibody.plant import AddMultibodyPlantSceneGraph
+from pydrake.multibody.tree import Joint_
+from pydrake.multibody.tree import JointIndex
+from pydrake.multibody.tree import RevoluteJoint
+from pydrake.multibody.tree import WeldJoint
+from pydrake.systems.framework import BasicVector_
+from pydrake.systems.framework import DiagramBuilder
+from pydrake.systems.framework import LeafSystem
+from pydrake.systems.framework import PublishEvent
+from pydrake.systems.framework import TriggerType
+from pydrake.systems.primitives import ConstantVectorSource
+
+from builtin_interfaces.msg import Time as TimeMsg
+from geometry_msgs.msg import TransformStamped
+import rclpy
+import rclpy.time
+from rclpy.qos import QoSProfile, DurabilityPolicy
+from std_msgs.msg import String as StringMsg
+from sensor_msgs.msg import JointState as JointStateMsg
+from tf2_ros import TransformBroadcaster
+
+
+# RuntimeError: Actuation input port for model instance iiwa must be connected.
+def no_control(plant, builder, model):
+    nu = plant.num_actuated_dofs(model)
+    u0 = numpy.zeros(nu)
+    constant = builder.AddSystem(ConstantVectorSource(u0))
+    builder.Connect(
+        constant.get_output_port(0),
+        plant.get_actuation_input_port(model))
+
+
+class SimulatorClock(LeafSystem):
```

Comment:
Works as intended, but `SimulatorClock` is unused in the demo. Using system time for the tf transforms allows restarting the demo over and over without restarting RViz. Using simulated time means RViz complains about out of date transforms. Maybe RViz could handle time jumps better.